pkg/codegen/go/gen.go

```diff
@@ -3244,6 +3245,18 @@ func GeneratePackage(tool string, pkg *schema.Package) (map[string][]byte, error
 	pathPrefix := packageRoot(pkg)
 
 	files := map[string][]byte{}
+
+	// Generate pulumiplugin.json
+	pulumiPlugin, err := (&plugin.PulumiPluginJSON{
+		Resource: true,
+		Name:     pkg.Name,
+		Server:   pkg.PluginDownloadURL,
+	}).JSON()
+	if err != nil {
+		return nil, fmt.Errorf("Failed to format pulumiplugin.json: %w", err)
+	}
+	files[path.Join(pathPrefix, "pulumiplugin.json")] = pulumiPlugin
```

Comment:
Good catch! To make matters even more complicated, `sdk/go/random` is a configurable option, it's not hardcoded. The current lookup method requires that `pulumiplugin.json` is in the same folder as `go.mod`, which makes sense to me. Except that it would mean putting another go specific file in the `sdk` folder. 

Let me think for a bit.

Reply:
After discussing offline, I intend to do the following
- Continue to generate `pulumiplugin.json` in `m.Dir/go/packagemane` by default.
- Unless `rootPackageName` or `importBasePath` is set. Then I will generate `pulumiplugin.json` in the top level directory. It will be the responsibility of the codegen user to move it next to `go.mod`.
- Change `pulumiplugin.json` lookup to probe
   - `m.Dir`
   - `m.Dir/go`
   - `m.Dir/go/${package name}`